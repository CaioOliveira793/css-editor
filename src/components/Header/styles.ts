import styled from 'styled-components';

export const Container = styled.header`
	display: flex;
	flex-flow: row nowrap;
	align-items: center;
	justify-content: space-between;

	width: 100%;
	padding: 24px 48px;
`;

export const PageTitle = styled.strong`
	font-size: 22px;
	font-weight: 500;
	letter-spacing: 2px;

	color: ${({ theme }) => theme.color.primaryText};
`;

export const Navigation = styled.nav`
	display: flex;
	flex-flow: row nowrap;
	align-items: center;
	justify-content: center;
	gap: 0px 18px;
`;

export const Anchor = styled.a`
	font-size: 12px;
	letter-spacing: 1px;
	text-transform: uppercase;
	text-decoration: none;
	cursor: pointer;
`;
