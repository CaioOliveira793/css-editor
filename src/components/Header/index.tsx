import React, { HTMLAttributes } from 'react';
import Link from 'next/link';

import { Container, Navigation, PageTitle, Anchor } from './styles';

interface HeaderProps extends HTMLAttributes<{}> { };

const Header: React.FC<HeaderProps> = ({ className }) => {
	return (
		<Container className={className}>
			<PageTitle>CSS Editor</PageTitle>
			<Navigation>
				<Link href="/properties">
					<Anchor>Properties</Anchor>
				</Link>
			</Navigation>
		</Container>
	);
}

export default Header;
