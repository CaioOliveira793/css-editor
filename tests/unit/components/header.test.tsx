import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import defaultTheme from '../../../src/styles/themes/default';

import Header from '../../../src/components/Header';

describe('Header', () => {

	beforeEach(() => {
		jest.mock('next/link', () => ({ children }) => children);
	});

	it('macth the snapshot', () => {
		const tree = renderer.create(
			<ThemeProvider theme={defaultTheme}>
				<Header links={[{ label: 'properties', href: '/properties' }]} />
			</ThemeProvider>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it(`have the 'CSS Editor' text`, () => {
		const { getByText } = render(
			<ThemeProvider theme={defaultTheme}>
				<Header links={[{ label: 'properties', href: '/properties' }]} />
			</ThemeProvider>
		);

		const mainText = getByText(/CSS Editor/);
		expect(mainText).toHaveTextContent(/CSS Editor/);
	});

	it('create a navigable link', () => {
		const { getByText } = render(
			<ThemeProvider theme={defaultTheme}>
				<Header links={[{ label: 'properties', href: '/properties' }]} />
			</ThemeProvider>
		);

		const anchor = getByText(/properties/i);
		expect(anchor).toHaveTextContent(/properties/i);
	});

	it('navigate with the link', async () => {
		const { getByText } = render(
			<ThemeProvider theme={defaultTheme}>
				<Header links={[{ label: 'properties', href: '/properties' }]} />
			</ThemeProvider>
		);

		const anchor = getByText(/properties/i);
		await waitFor(() => fireEvent.click(anchor));

		const propertiesPage = getByText('chose a item');
		expect(propertiesPage).toHaveTextContent('chose a item');
	});

});
