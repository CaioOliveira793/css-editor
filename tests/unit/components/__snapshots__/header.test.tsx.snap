// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header macth the snapshot 1`] = `
<header
  className="styles__Container-sc-1qfn4k8-0 bHEOWO"
>
  <strong
    className="styles__PageTitle-sc-1qfn4k8-1 jckDPd"
  >
    CSS Editor
  </strong>
  <nav
    className="styles__Navigation-sc-1qfn4k8-2 jUgoPa"
  >
    <a
      className="styles__Anchor-sc-1qfn4k8-3 dkiYpB"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      properties
    </a>
  </nav>
</header>
`;
